src/flags/recursion.rs

```diff
@@ -35,25 +34,18 @@ impl Recursion {
     /// - [enabled_from_arg_matches](Recursion::enabled_from_arg_matches)
     /// - [enabled_from_config](Recursion::enabled_from_config)
     /// - [Default::default]
-    ///
-    /// # Note
-    ///
-    /// The configuration file's Yaml is read in any case, to be able to check for errors and print
-    /// out warnings.
     fn enabled_from(matches: &ArgMatches, config: &Config) -> bool {
-        let mut result: bool = Default::default();
-
-        if config.has_yaml() {
-            if let Some(value) = Self::enabled_from_config(config) {
-                result = value;
+        if let Some(recursion) = &config.recursion {
+            if let Some(enabled) = recursion.enabled {
+                return enabled;
```

Comment:
This method breaks the previously agreed on precedence. Command line options should always override the config file options.